feat(dashboard): Nv 5674 dashboard cards

BiswaViraj opened the pull request:
### What changed? Why was the change needed?
Related maily PR: https://github.com/novuhq/maily.to/pull/19 which is needed to make the button go fullwidth in one of the cards
<!-- Also include any relevant links, such as Linear tickets, Slack discussions, or design documents. -->

### Screenshots

https://github.com/user-attachments/assets/26abddb1-6657-4ac6-bf18-cf1d7679fa71


<!-- If the changes are visual, include screenshots or screencasts. -->

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Related enterprise PR

<!-- A link to a dependent pull request  -->

### Special notes for your reviewer

<!-- Specific instructions or considerations you want to highlight for the reviewer. -->

</details>

BiswaViraj committed:
feat(email-editor): add new card components and integrate into editor blocks

BiswaViraj committed:
feat(email-editor): replace PNG previews with WEBP format for improved performance

BiswaViraj commented on apps/dashboard/public/images/email-editor/check-icon.svg:
the png/webp version was too blurry, so kept the svg

BiswaViraj committed:
fix(cards): correct description text for card component

BiswaViraj committed:
feat(cspell): add "automators" to custom words list

BiswaViraj committed:
Merge branch 'nv-5657-dashboard-enhance-the-variable-pill' into nv-5674-dashboard-cards

LetItRock commented on apps/dashboard/src/components/workflow-editor/steps/email/blocks/cards.tsx:
```diff
@@ -0,0 +1,347 @@
+                  content: [
+                    {
+                      type: 'text',
+                      text: 'Your free trial for Novu Business Events and 1 more product with Novu US, Inc. will end soon. You have an upcoming payment on',
```
```suggestion
                      text: 'Your free trial for Novu Business Events and 1 more product with Novu US, Inc. will end soon. You have an upcoming payment on ',
```

LetItRock commented on apps/dashboard/src/components/workflow-editor/steps/email/blocks/cards.tsx:
```diff
@@ -0,0 +1,347 @@
+                    borderRadius: 'smooth',
+                    buttonColor: '#f8f8f8',
+                    textColor: '#141313',
+                    showIfKey: null,
```
please make the button smaller and maybe add some margin to move it to the bottom, but margin is not supported so you would have to update it in the maily pr: 
```
paddingBottom: 6,
paddingLeft: 24,
paddingRight: 24,
paddingTop: 6,
marginTop: 10,
```

LetItRock replied:
please adjust all the buttons

LetItRock commented on apps/dashboard/src/components/workflow-editor/steps/email/blocks/cards.tsx:
```diff
@@ -0,0 +1,347 @@
+          content: [
+            {
+              type: 'columns',
+              attrs: { showIfKey: null, gap: 0 },
```
there should be more padding, but it doesn't work on a section, so we would have probably to play with the paddings in columns

LetItRock commented:
it would be nice if we can position all these card hover images so they are fully visible in the popover
![Screenshot 2025-04-16 at 09 16 07](https://github.com/user-attachments/assets/2bf1dcd4-226c-40db-b29d-fcb8011a0278)

BiswaViraj committed:
Merge branch 'nv-5657-dashboard-enhance-the-variable-pill' into nv-5674-dashboard-cards

BiswaViraj committed:
Merge branch 'nv-5657-dashboard-enhance-the-variable-pill' into nv-5674-dashboard-cards

BiswaViraj committed:
fix(cards): add missing space in trial payment notification text

BiswaViraj committed:
fix(cards): adjust spacing and padding for horizontal and information cards

BiswaViraj committed:
feat(cards): add BlockCustomPreview component for enhanced card previews

BiswaViraj committed:
fix(dependencies): update @maily-to/core-digest to version 0.2.7-novu.11-core

BiswaViraj committed:
fix(cards): refactor preview rendering to use BlockCustomPreview for horizontal card

BiswaViraj committed:
Merge branch 'nv-5657-dashboard-enhance-the-variable-pill' into nv-5674-dashboard-cards

BiswaViraj committed:
feat(telemetry): add CARD_BLOCK_ADDED event for card block tracking
fix(cards): update telemetry tracking to use CARD_BLOCK_ADDED for card commands

BiswaViraj committed:
fix(editor): optimize highlight blocks creation for enhanced digest
